src/core/gltf2exporter/gltf2exporter_p.cpp

```diff
@@ -0,0 +1,716 @@
+
+    QJsonObject m_root;
+    QJsonArray m_buffers;
+    QJsonArray m_bufferViews;
+    QJsonArray m_accessors;
+    QJsonArray m_meshes;
+    QJsonArray m_images;
+
+    QByteArray m_compressedBuffer;
+    const int m_compressedBufferIndex;
+    int m_newBufferViewIndex;
+    std::set<int> m_accessorsToClean;
+    QDir m_basePath, m_destination;
+
+    const GLTF2ExportConfiguration &m_conf;
+    const GLTF2Import::GLTF2Context &m_context;
+
+    // Compress a single mesh
+    QJsonObject compressMesh(QJsonObject mesh_json, int mesh_idx)
+    {
+        const auto &primitives = m_context.mesh(mesh_idx).meshPrimitives;
+        auto primitives_json = mesh_json[GLTF2Import::KEY_PRIMITIVES].toArray();
+
+        Q_ASSERT_X(primitives_json.size() == primitives.size(), "GLTF2DracoCompressor::compressMesh", "Bad primitive count");
+
+        for (int p = 0; p < primitives_json.size(); ++p) {
+            auto primitive_json = primitives_json[p].toObject();
+
+            // First check if the primitive is already draco-compressed
+            auto ext_it = primitive_json.find(GLTF2Import::KEY_EXTENSIONS);
+            if (ext_it != primitive_json.end()) {
+                auto ext = ext_it->toObject();
+                if (ext.contains(GLTF2Import::KEY_KHR_DRACO_MESH_COMPRESSION_EXTENSION)) {
+                    continue;
+                }
+            }
+
+            auto compressed_mesh = compressPrimitive(
+                    primitive_json,
+                    primitives[p]);
+
+            if (!compressed_mesh.primitiveJson.empty()) {
+                //* all the side-effects of this method are in this block: */
+                primitives_json[p] = compressed_mesh.primitiveJson;
+                m_compressedBuffer.push_back(compressed_mesh.compressedData);
+                m_bufferViews.push_back(compressed_mesh.newBufferView);
+                m_accessorsToClean.insert(compressed_mesh.accessorsToClean.begin(), compressed_mesh.accessorsToClean.end());
+
+                m_newBufferViewIndex++;
+            } else {
+                const auto meshName = mesh_json[GLTF2Import::KEY_NAME].toString();
+                if (!meshName.isEmpty())
+                    m_errors << QStringLiteral("A mesh could not be compressed: %1 -> %2").arg(meshName).arg(p);
+                else
+                    m_errors << QStringLiteral("A mesh could not be compressed: %1 -> %2").arg(mesh_idx).arg(p);
+            }
+        }
+
+        mesh_json[GLTF2Import::KEY_PRIMITIVES] = std::move(primitives_json);
+        return mesh_json;
+    }
+
+    // Compress a single primitive of a mesh
+    CompressedGLTFPrimitive compressPrimitive(
+            QJsonObject primitive_json, const Kuesa::GLTF2Import::Primitive &primitive) const
+    {
+        CompressedGLTFPrimitive compressed_primitive;
+
+        // Do the compression
+        const auto compressed = Kuesa::DracoCompression::compressMesh(*primitive.primitiveRenderer->geometry(), m_conf);
```

Comment:
this fails to compile with clang if the destructor is added to the `CompressedMesh` class (which disables the generation of the default copy constructor)